src/com/ensa/gi4/service/impl/GestionMaterielFacadeImpl.java

```diff
@@ -0,0 +1,180 @@
+
+import com.ensa.gi4.modele.Materiel;
+import com.ensa.gi4.modele.TypeMateriel;
+import com.ensa.gi4.modele.User;
+import com.ensa.gi4.service.api.GestionMaterielService;
+import com.ensa.gi4.service.api.GestionMaterielServiceFacade;
+import com.ensa.gi4.service.api.GestionUserService;
+
+@Service
+public class GestionMaterielFacadeImpl implements GestionMaterielServiceFacade {
+
+	@Autowired
+	@Qualifier("gestionLivreService")
+	private GestionMaterielService gestionLivreService;
+	@Autowired
+	@Qualifier("gestionChaiseService")
+	private GestionMaterielService gestionChaiseService;
+	@Autowired
+	private GestionUserService gestionUserService;
+
+	@Override
+	public List<Materiel> afficherMateriel(TypeMateriel type) {
+		final GestionMaterielService serviceActuel;
+		switch (type) {
+		case LIVRE:
+			serviceActuel = gestionLivreService;
+			break;
+		case CHAISE:
+			serviceActuel = gestionChaiseService;
+			break;
+		default:
+			System.out.println("Le type " + type + " n'est pas reconnu");
+			return null;
+		}
+		List<Materiel> materiels = serviceActuel.findAll();
+		return materiels;
+	}
+
+	@Override
+	public void ajouterNouveauMateriel(TypeMateriel type, Materiel materiel) {
+		final GestionMaterielService serviceActuel;
+		switch (type) {
+		case LIVRE:
+			serviceActuel = gestionLivreService;
+			break;
+		case CHAISE:
+			serviceActuel = gestionChaiseService;
+			break;
+		default:
+			System.out.println("Le type " + type + " n'est pas reconnu");
+			return;
+		}
+		serviceActuel.add(materiel);
+	}
+
+	@Override
+	public void supprimerMateriel(TypeMateriel type, int id) {
+		final GestionMaterielService serviceActuel;
+		switch (type) {
+		case LIVRE:
+			serviceActuel = gestionLivreService;
+			break;
+		case CHAISE:
+			serviceActuel = gestionChaiseService;
+			break;
+		default:
+			System.out.println("Le type " + type + " n'est pas reconnu");
+			return;
+		}
+		serviceActuel.delete(id);
+	}
+
+	@Override
+	public Materiel findOne(TypeMateriel type, String nom) {
+		final GestionMaterielService serviceActuel;
+		switch (type) {
+		case LIVRE:
+			serviceActuel = gestionLivreService;
+			break;
+		case CHAISE:
+			serviceActuel = gestionChaiseService;
+			break;
+		default:
+			System.out.println("Le type " + type + " n'est pas reconnu");
+			return null;
+		}
+		Materiel materiel = serviceActuel.findOne(nom);
+		return materiel;
+	}
+
+	@Override
+	public void modifierMateriel(TypeMateriel type, int id, Materiel materiel) {
+		final GestionMaterielService serviceActuel;
+		switch (type) {
+		case LIVRE:
+			serviceActuel = gestionLivreService;
+			break;
+		case CHAISE:
+			serviceActuel = gestionChaiseService;
+			break;
+		default:
+			System.out.println("Le type " + type + " n'est pas reconnu");
+			return;
+		}
+		serviceActuel.update(id, materiel);
+	}
+
+	@Override
+	public void louerMateriel(TypeMateriel type, String nomMateriel, int userId) {
+		final GestionMaterielService serviceActuel;
+		switch (type) {
```

Comment:
```java 
switch (type) {
		case LIVRE:
			serviceActuel = gestionLivreService;
			break;
		case CHAISE:
			serviceActuel = gestionChaiseService;
			break;
		default:
			System.out.println("Le type " + type + " n'est pas reconnu");
			return;
		}
```
this duplicate code can be refactored

Reply:
D'accord Monsieur merci pour vos remarques je vais les prendre en considération